.github/workflows/test-images.yml

```diff
@@ -0,0 +1,130 @@
+name: Test images manager
+
+on:
+  workflow_run:
+    workflows: [Trigger test images]
+    types: [completed]
+
+env:
+  REPOSITORY: quay.io/pelorus
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      pr_type: ${{ steps.variables.outputs.pr_type }}
+      pr_number: ${{ steps.variables.outputs.pr_number }}
+      commit_hash: ${{ steps.variables.outputs.commit_hash }}
+    steps:
+      - name: Download trigger information
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: trigger-test-images.yml
+          run_id: ${{ github.event.workflow_run.id }}
+          name: trigger-information
+          path: ./
+
+      # TODO remove!!!
+      - name: DEBUG
+        run: cat trigger-information.json
+
+      - name: Store variables
+        id: variables
+        run: |
+          echo pr_type=$(cat trigger-information.json | jq '.pr_type') >> "$GITHUB_OUTPUT"
+          echo pr_number=$(cat trigger-information.json | jq '.pr_number') >> "$GITHUB_OUTPUT"
+          echo commit_hash=$(cat trigger-information.json | jq '.commit_hash') >> "$GITHUB_OUTPUT"
+
+  # TODO
+  # delete-images:
+  #   runs-on: ubuntu-latest
+  #   if: ${{ needs.setup.outputs.pr_type == 'closed' }}
+  #   needs: setup
+  #   steps:
+  #     - name: Delete test images
+  #       run: |
+  #         podman search --list-tags $REPOSITORY/rc-pelorus-exporter
+  #         podman search --list-tags $REPOSITORY/rc-pelorus-operator
+  #         podman search --list-tags $REPOSITORY/rc-pelorus-operator-bundle
+  #         delete all that have PR${{ needs.setup.outputs.pr_number }} in name...
+
+  create-images:
+    runs-on: ubuntu-latest
+    if: ${{ needs.setup.outputs.pr_type != 'closed' }}
+    needs: setup
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Build exporter test image
+        id: build-exporter
+        uses: redhat-actions/s2i-build@v2
+        with:
+          image: rc-pelorus-exporter
+          path_context: 'exporters'
+          builder_image: 'registry.access.redhat.com/ubi8/python-39'
+          tags: PR${{ needs.setup.outputs.pr_number }}-${{ needs.setup.outputs.commit_hash }}
+
+      - name: Push exporter test image to Quay.io
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-exporter.outputs.image }}
+          tags: ${{ steps.build-exporter.outputs.tags }}
+          registry: $REPOSITORY
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Log in to Quay.io
+        uses: redhat-actions/podman-login@v1
+        with:
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+          registry: quay.io
+
+      - name: Use test images in operator
```

Comment:
I still do not know how we will test this PR, but I feel this is the most important step @mpryc 

Can you check if everything looks ok

Reply:
that is not the idea

the idea is when you open (or update) an PR that touches one of the 3 folders, it triggers this workflow

I do not know how to make workflow run on commands like Prow

Reply:
You are looking into the following part:

https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/oc/openshift-oc-master-presubmits.yaml#L209

And then inside the job some logic to wait for the containers to be available on the quay. You know the hash of the last change as prow has the source code, so you should be expecting similar TAG on the quay images.

Reply:
the idea is that the test images have tag `PR{pr number}-{commit hash}`

looking into Prow, this should be achievable with env vars https://docs.prow.k8s.io/docs/jobs/#job-environment-variables (`PULL_NUMBER` and `PULL_PULL_SHA`)

I think sha from env var of Prow is short end from github actions is long, but I think we can address this kind of issue in the corresponding PR. This one is just for create the test images, not changing e2e tests yet